README_production.rst

```diff
@@ -0,0 +1,293 @@
+
+
+       $ sudo python3 -m yaksh.code_server # For Python 3.x
+
+   Put this in the background once it has started since this will not
+   return back the prompt. It is important that the server be running
+   *before* students start attempting the exam. Using sudo is necessary
+   since the server is run as the user "nobody". This runs the number
+   ports configured in the settings.py file in the variable
+   "N\_CODE\_SERVERS". The "SERVER\_TIMEOUT" also can be changed there.
+   This is the maximum time allowed to execute the submitted code. Note
+   that this will likely spawn multiple processes as "nobody" depending
+   on the number of server ports specified.
+
+   You can also use a Dockerized code server, see :ref:`dockerized-code-server`.
+
+4. Now, Run:
+
+   ::
+
+          python manage.py runserver <desired_ip>:<desired_port>
+
+5. Create a Superuser/Administrator:
+
+   ::
+
+       python manage.py createsuperuser
+
+6. Go to http://desired\_host\_or\_ip:desired\_port/exam
+
+   And you should be all set.
+
+7. Note that the directory "output" will contain directories, one for
+   each user. Users can potentially write output into these that can be
+   used for checking later.
+
+8. As admin user you can visit http://desired\_host\_or\_ip/exam/monitor to view results
+   and user data interactively. You could also "grade" the papers
+   manually if needed.
+
+.. _dockerized-code-server:
+
+Using Dockerized Code Server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Install
+   `Docker <https://docs.docker.com/engine/installation/>`__
+
+2. Go to the directory where the project is located cd
+   /path/to/online\_test
+
+3. Create a docker image. This may take a few minutes docker build -t
+   yaksh\_code\_server -f ./docker/Dockerfile\_codeserver
+
+4. Check if the image has been created using the output of, docker
+   images
+
+5. Run the invoke script using the command ``invoke start`` The command
+   will create and run a new docker container (that is running the
+   code\_server.py within it), it will also bind the ports of the host
+   with those of the container
+
+6. You can use ``invoke --list`` to get a list of all the available commands
+
+.. _deploying-multiple-dockers:
+
+######################################
+Deploying Multiple Dockers
+######################################
+
+Follow these steps to deploy and run the Django Server, MySQL instance and Code Server in seperate Docker instances.
+
+1. Install `Docker <https://docs.docker.com/engine/installation/>`__
+
+2. Install `Docker Compose <https://docs.docker.com/compose/install/>`__
+
+2. Go to the ``docker`` directory where the project is located:
+   
+   ::
+
+       cd /path/to/online_test/docker
+
+3. Build the docker images
+
+   ::
+
+       invoke build
+
+4. Run the containers and scripts necessary to deploy the web
+   application
+
+   ::
+
+       invoke deploy
+
+5. Use ``invoke deploy --fixtures`` to load the fixtures
```

Comment:
`invoke deploy` is a compulsory command to run. So to connect to database it is ok. but invoke deploy --fixtures should be changed to `invoke loadfixtures` because the above command unnecessarily does everything which `invoke deploy` does. So I feel it should be changed.